java-checks/src/main/java/org/sonar/java/checks/UtilityClassWithPublicConstructorCheck.java

```diff
@@ -80,17 +80,22 @@ private static boolean extendsAnotherClassOrImplementsSerializable(ClassTree cla
   }
 
   private static boolean hasOnlyStaticMembers(ClassTree classTree) {
-    if (classTree.members().isEmpty()) {
+    List<Tree> members = classTree.members();
+    if (members.isEmpty() || noStaticMember(members)) {
```

Comment:
I'm not really convinced. the `or` operator is short-circuit, so if members are empty, we won't call `noStaticMember`. I also would like to avoid collecting and creating a new list.